### PR TITLE
[SOFT-61] Implement fault bps

### PR DESCRIPTION
**Pull request by g-jessmuir**
Add a `fault_bps` module rather than defining it in the `bms.h` header. Also expose a `relay_fault` function from the relays. `fault_bps` effectively wraps `relay_fault`, but does the bitset accounting.

### CODE REVIEW

File: projects/bms_carrier/src/fault_bps.c [hunk 1]
```diff
@@ -1,7 +1,22 @@
 #include "bms.h"
+#include "exported_enums.h"
```
[the remark this answers is not in the document]
Comment (g-jessmuir) [reply projects/bms_carrier/src/fault_bps.c:4]: added

File: projects/bms_carrier/src/relay_sequence.c [hunk 2]
```diff
@@ -23,16 +24,19 @@ void prv_relay_fault(RelayStorage *storage) {
   // skip assertion, we're already faulted
   soft_timer_cancel(storage->assertion_timer_id);
   soft_timer_cancel(storage->next_step_timer_id);
-  // fault bps
-  fault_bps(EE_BPS_STATE_FAULT_RELAY, false);
+  if (internal) {
+    // fault bps
+    fault_bps(EE_BPS_STATE_FAULT_RELAY, false);
```
[the remark this answers is not in the document]
Comment (g-jessmuir) [reply projects/bms_carrier/src/relay_sequence.c:29]: whoops you're right, added that check

File: projects/bms_carrier/src/relay_sequence.c [hunk 3]
```diff
@@ -23,16 +24,19 @@ void prv_relay_fault(RelayStorage *storage) {
...
+  }
+  CAN_TRANSMIT_BATTERY_RELAY_STATE(storage->hv_enabled, storage->gnd_enabled);
```
[the remark this answers is not in the document]
Comment (g-jessmuir) [reply projects/bms_carrier/src/relay_sequence.c:31]: the can transmit battery relay state is for telemetry purposes; setting the relay states is done through power main sequence messages

[the remark this answers is not in the document]
Comment (GeraldNDA) [reply projects/bms_carrier/src/relay_sequence.c:31]: ah, I see

**Review by GeraldNDA — APPROVED** [review]
LGTM

**Pull request merged**